Dockerfile

```diff
@@ -0,0 +1,38 @@
+# Prerequisites for using dockerfile can be seen here https://github.com/zephyrtronium/robot/issues/76
+
+# Main image
+FROM golang:1.23-alpine AS build
+
+# Define config
+ARG CONFIG
+
+# Copy repo and populate Go resources
+COPY . .
+RUN go mod download
+
+# Build Robot
+RUN --mount=type=cache,target=/root/.cache/go-build \
+  --mount=type=cache,target=/go/pkg \
+  CGO_ENABLED=0 \
+  go build -o /build/robot github.com/zephyrtronium/robot
+
+# Prepare minimised image
+FROM alpine
+
+# Set config
+ARG CONFIG
+ENV CONFIG=${CONFIG:-robot.toml}
+
+# Copy Robot binary and required resources
+COPY --from=build /build/robot /bin/robot
+COPY robot_key /
+COPY secret /
+
+# Copy config file as robot.toml
+COPY ${CONFIG} /robot.toml
```

Comment:
Not really familiar with the built structure of robot. There might be a better place to house these that robot can see in its current directory to make specifying these in the config a bit simpler

Reply:
The secret key and client secret could be taken as build arguments. Alternatively, since those config options have environment variables expanded, it should be possible to take advantage of Docker's secrets mechanism. I don't know the details, but it should be similar to what the example config does for systemd secrets.